src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -38,15 +38,16 @@ internal static partial class CapiHelper
         public static string UpgradeDSS(int dwProvType, string wszProvider)
         {
             string wszUpgrade = null;
-            SafeProvHandle safeProvHandle = SafeProvHandle.InvalidHandle;
             if (string.Equals(wszProvider, MS_DEF_DSS_DH_PROV, StringComparison.Ordinal))
             {
+                SafeProvHandle safeProvHandle;
                 // If this is the base DSS/DH provider, see if we can use the enhanced provider instead.
-                if (S_OK == AcquireCryptContext(ref safeProvHandle, null, MS_ENH_DSS_DH_PROV, dwProvType,
+                if (S_OK == AcquireCryptContext(out safeProvHandle, null, MS_ENH_DSS_DH_PROV, dwProvType,
                                         (uint)CryptAcquireContextFlags.CRYPT_VERIFYCONTEXT))
                 {
                     wszUpgrade = MS_ENH_DSS_DH_PROV;
                 }
+                safeProvHandle.Dispose();
```

Comment:
Well, in the case of this particular one, that wouldn't really work; since this is doing an opportunistic call.

I agree that there's virtue in that approach, and in the CNG library I'd agree that it was the right thing to do (because that library will continue to evolve).  Here I'm less worried about someone getting things wrong in the future.

I'd be okay with it as a separate commit/PR, but it seems pretty low-priority.
I am, of course, prepared to eat these words in the future :).